### PR TITLE
Grand cleanup in groups endpoints.

### PR DESCRIPTION
**Pull request by krulis-martin**
Instances groups endpoint and all groups endpoints removed, default groups endpoint updated to support filters.

### CODE REVIEW

File: app/model/repository/Groups.php [hunk 1]
```diff
@@ -103,13 +106,86 @@ public function findByName(string $locale, string $name, Instance $instance, ?Gr
...
+   */
+  public function groupsAncestralClosure(array $groups)
+  {
+    $res = [];
+    foreach ($groups as $group) {
+      $groupId = $group->getId();
+      if (array_key_exists($groupId, $res)) continue;
+      $res[$groupId] = $group;
+      foreach ($group->getParentGroupsIds() as $id) {
+        if (array_key_exists($id, $res)) break;
```
Comment (Neloop) [inline app/model/repository/Groups.php]: Where in the name of our lord and savior are braces?

File: app/model/repository/Groups.php [hunk 2]
```diff
@@ -103,13 +106,86 @@ public function findByName(string $locale, string $name, Instance $instance, ?Gr
...
+   * Gets an initial set of groups and produces a set of groups which have the
+   * original set as a subset and every group has its parent in the set as well.
+   * @param Group[] $groups Initial set of groups
+   */
+  public function groupsAncestralClosure(array $groups)
+  {
+    $res = [];
+    foreach ($groups as $group) {
+      $groupId = $group->getId();
+      if (array_key_exists($groupId, $res)) continue;
```
Comment (Neloop) [inline app/model/repository/Groups.php]: same as below

File: app/model/repository/Groups.php (hunk 1 above)
Comment (krulis-martin) [reply app/model/repository/Groups.php]: As a convention, I do not write them in case of trivial statements (like if-break, if-return).

Comment (Neloop) [reply app/model/repository/Groups.php]: Yeah... as a convention in our api, we write them everywhere

File: app/model/repository/Groups.php [hunk 3]
```diff
@@ -170,11 +170,14 @@ public function groupsAncestralClosure(array $groups)
     $res = [];
     foreach ($groups as $group) {
       $groupId = $group->getId();
-      if (array_key_exists($groupId, $res)) continue;
+      if (array_key_exists($groupId, $res)) { // @neloop made me do that
```
Comment (Neloop) [inline app/model/repository/Groups.php]: you are god damn right

**Pull request merged**